runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -604,10 +614,20 @@ private void rebuildTargetList()
 	}
 
 	private void setTask(String name, int amt, int initAmt)
+	{
+		setTask(name, amt, initAmt, "");
+	}
+
+	private void setTask(String name, int amt, int initAmt, String location)
 	{
 		taskName = name;
 		amount = amt;
 		initialAmount = initAmt;
+		if (location == null)
+		{
+			location = "";
```

Comment:
Why?

Reply:
To simplify the logic where Matcher.group(String) is used, for the Pattern group that selects the task location, which is not always present. Matcher.group(String) returns null if the group is not present, so I sanitize null into the empty string here.